### ISSUE TITLE
need nested output dir

### ISSUE THREAD

**Issue by zyy7259** [postcss/postcss-cli#30]
I figure out that I can use glob expression in the 'input' field, but no corresponding nested folders generated in the output dir. Here is my configuration:

```
module.exports = {
  input: 'src/postcss/**/*.css',
  dir: 'dist/css',
  use: [
    'precss',
    'postcss-custom-properties',
    'postcss-calc',
    'autoprefixer',
    'cssnano'
  ],
  map: 'file',
  'local-plugins': true,
  watch: true,
  autoprefixer: {
    browsers: ['Android >= 4', 'iOS >= 7', 'Chrome >= 10', 'Firefox >= 10', 'IE >= 8']
  }
}
```

**Comment by vbalcom** [comment]
Hi there, I've come across a similar need for nested output folders. The problem comes from the compile function, where the destination `--dir` and input basename are combined  (`output = path.join(argv.dir, path.basename(input));`), thus ignoring any nested folders, or overwriting any files with the same name. 

We use index.css as an entry point for all our component css files, so we were having some trouble.
I would recommend adding in an option such as `--context` or `--cwd` that gives the parent path before an input glob (similar to how gulp and grunt handle nested directories). That way we can tell compile to use the input's pathname after the context as the output, therefore mimicking any nested directories. The feature would be optional, to be used when you are using `--dir`.
My config would look like such:
```json
{
  "use": [ ... ],
  "input": "/**/index.css",
  "context": "library/styles"
  "dir": "library/dist",
}
```
If that seems like an agreeable way of going about this, I'd be happy to submit a pull request.

**Comment by shirleyMHao** [comment]
@vbalcom , i use the context，but it doesn't work.
input directory structure like
`base
  -- css
       ----1.css
       ----testfolder
             ----2.css
       ----testfolder2
             ----3.css`
I expect :
`dist
  -- css
       ----1.css
       ----testfolder
             ----2.css
       ----testfolder2
             ----3.css`
my postcss.config.js 
`
but it seems to process the root directory,not relative of the config file exits.
module.exports={
    input: '/**/*.css',
    dir: 'dist/css',
    context: 'css'
    use:[
        'postcss-unprefix',
        'postcss-flexboxfixer',
        'autoprefixer',
    ],  
    autoprefixer:{
        browsers:['defaults','IE>=8']
    }   
}
`

**Comment by vbalcom** [comment]
@shirleyMHao the context will always be the path to the folder from the root. So for your project structure, you would use `context: 'base/css'`

**Comment by shirleyMHao** [comment]
@vbalcom , I change the context to "base/css", but postcss-cli just do not print any error，and can't finish. I   don't know what is wrong.

**Comment by vbalcom** [comment]
@shirleyMHao You'll need to be on the correct fork where this feature is implemented: (https://github.com/postcss/postcss-cli/pull/31)

It looks like they'll be supporting context in the next version of postcss-cli as well.

**Comment by michael-ciniawsky** [comment]
@vbalcom @zyy7259  👋  Just to clearify, you're 'usecase' is e.g

```
| – css
|     | – nav
|     |     | – index.css
|     | – header
|     |     | – index.css 
| – dist
|     | – nav
|     |     | – result.css
|     | – header
|     |     | – result.css  
```

```bash
postcss css/**/*.css -d dist/
```

This should preserve the directory structure `css/**/*.css // => **: ['nav' , 'header']` from the input  in the dist right? Or did I missunderstand something 🙃 ?

**Comment by bfred-it** [comment]
@michael-ciniawsky the double-star globbing is not recursive in bash 3.2 (on my OS X 10.11) and even then postcss doesn't seem to keep the relative folder structure. So this is the output of your command:

```
| – css
|    | – nav
|    |   | – index.css
|    | – header
|    |   | – index.css 
| – dist
|    | – index.css  
```

It'd be great if `postcss-cli` would support `input` to be a folder like `babel-cli` does:

```sh
babili srcDir --out-dir distDir
```

**Comment by RyanZim** [comment]
@bfred-it postcss-cli processes globs itself, to get double-star globbing working, you have to quote your glob. Anyhow, it won't keep relative folder structure anyhow.

I sorta like the idea of an input folder; @michael-ciniawsky thoughts on this? @bfred-it Would you be willing to do a PR for this?

**Comment by swese44** [comment]
we also need this feature, glad to see the PR merged. when can we expect this to be published? thanks!

**Comment by RyanZim** [comment]
@swese44 As soon as I get a chance. Hopefully this evening.

**Comment by swese44** [comment]
cool thanks @RyanZim!

**Comment by RyanZim** [comment]
@swese44 Finally got it published as v3.2.0 tonight.

**Comment by swese44** [comment]
thanks all! works perfectly
